lib/created_message_queue_handler.ts

```diff
@@ -247,15 +261,36 @@ export async function handleMessage(
   // query succeeded, we may have a profile
   const maybeProfile = errorOrMaybeProfile.value;
 
-  // whether the recipient wants us to store the message content
-  const isMessageStorageEnabled = maybeProfile.exists(
-    profile => profile.isInboxEnabled === true
+  // channels ther user has blocked for this sender service
+  const blockedChannels = maybeProfile
+    .chain(profile =>
+      fromNullable(profile.blockedChannels).map(
+        (bc: IProfileBlockedChannels) =>
+          bc[newMessageWithContent.senderServiceId]
+      )
+    )
+    .getOrElse(new Set());
+
+  winston.debug(
+    "handleMessage|Blocked Channels(%s): %s",
+    newMessageWithContent.fiscalCode,
+    JSON.stringify(blockedChannels)
   );
 
   //
   //  Inbox storage
   //
 
+  // check if the user has blocked inbox message storage from this service
+  const userHasBlockedMessageStorage = blockedChannels.has(
```

Comment:
just to be clear (while reading the code I forgot this is for a specific service) `userHasBlockedMessageStorageForService`